### PR TITLE
Add mikeneko65 keyboard

### PR DESCRIPTION
**Pull request by takishim**
## Description

Add mikeneko65 keyboard to qmk.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/mikeneko65/rev3/config.h [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x1028
```
Comment (drashna) [inline keyboards/mikeneko65/rev3/config.h:22]: This Vendor ID is in use, and needs to be changed. 

https://www.the-sz.com/products/usbid/index.php?v=0x1028&p=&n=

Anything above 0x4000 should be okay.

File: keyboards/mikeneko65/rev3/readme.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Mikeneko 65 V3
+
+A simple 65% keyboard
+
+* Keyboard Maintainer: [takishim](https://github.com/takishim)
+* Hardware Supported: Mikeneko 65 V4
+* Hardware Availability: [Open source on GitHub](https://github.com/takishim/mikeneko65/tree/v3)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/mikeneko65/rev3/readme.md:9]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/mikeneko65/rev4/config.h [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID    0x1028
```
Comment (drashna) [inline keyboards/mikeneko65/rev4/config.h:22]: As above.

File: keyboards/mikeneko65/rev4/readme.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Mikeneko 65 V4
+
+A simple 65% keyboard
+
+* Keyboard Maintainer: [takishim](https://github.com/takishim)
+* Hardware Supported: Mikeneko 65 V4
+* Hardware Availability: [Open source on GitHub](https://github.com/takishim/mikeneko65/tree/v4)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/mikeneko65/rev4/readme.md:9]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by kinmojr** [comment]
Thank you for reviewing.
I fixed Vendor ID and added an explanation of how to enter the bootloader.
Could you please check it ?

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**